File: springboot-java-backend/src/main/java/com/example/template/routes/HealthController.java
package com.example.template.routes;

import com.example.template.state.ReadinessState;
import com.example.template.types.SuccessResponse;
import java.util.Map;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HealthController {
  private final ReadinessState readinessState;

  public HealthController(ReadinessState readinessState) {
    this.readinessState = readinessState;
  }

  @GetMapping("/health")
  public SuccessResponse<Map<String, Object>> health() {
    return SuccessResponse.of(Map.of("status", "ok", "ready", readinessState.isReady()));
  }
}

File: laravel-php-backend/app/Http/Controllers/HealthController.php
<?php

declare(strict_types=1);

namespace App\Http\Controllers;

use App\State\Readiness;
use App\Types\Api;

final class HealthController
{
    public function health(): array
    {
        return Api::success(['status' => 'ok']);
    }

    public function ready(): array
    {
        if (!Readiness::isReady()) {
            http_response_code(503);
            return [
                'type' => 'https://api.example.dev/errors/service-unavailable',
                'title' => 'Service Unavailable',
                'status' => 503,
                'detail' => 'Service not ready',
                'instance' => '/ready',
                'code' => 'SERVICE_UNAVAILABLE',
                'requestId' => request_context()->requestId,
                'traceId' => request_context()->traceId,
                'errors' => [],
            ];
        }

        return Api::success(['status' => 'ready']);
    }
}
